src/compile/cache/mod.rs

```diff
@@ -51,45 +67,68 @@ pub fn cache_dir() -> Result<PathBuf, CacheDirNotSupported> {
 	Ok(env::temp_dir().join("cairo-foundry-tests"))
 }
 
-fn read_cache_file(path: &PathBuf) -> Result<Cache, CacheError> {
-	let file = read_to_string(path)?;
-	let data = serde_json::from_str::<Cache>(file.as_str())?;
-	Ok(data)
+// TODO: support cashing compile errors to avoid recompiling files that failed before
+#[derive(Serialize, Deserialize, Debug)]
+pub struct CompileCacheItem {
+	pub hash: u64,
+	// TODO: make program_json: ProgramJson, we're not using it right now because
+	// it doesn't implement Serialize
```

Comment:
Do you want to do a PR to lambdaclass/cairo-rs to make it implement Serialize yourself?

Reply:
@tdelabro Yes, I'd like to. How to proceed ? Should I just open a PR in lambdaclass/cairo-rs project ?

Reply:
Yep, Open a PR on their repo.
If you want, make it a draft first and send it to me. I will review it first and then you can ask them for the final review.

Reply:
@tdelabro it turns out making `ProgramJson` serializable involves much more changes than what I expected, I created an issue rather than directly a PR on their repo, here it is: https://github.com/lambdaclass/cairo-rs/issues/856, should I wait for someone to agree it's a good feature to implement and assign it to me or not ?

Reply:
@haroune-mohammedi Wait for them to answer and validate it's something we want and maybe comment on the design. 
Do you have a telegram contact you want to share with me, so I can add you to the telegram dedicated to rust and starknet? Lambda class team is really active on this telegram channel

Reply:
@tdelabro Yes, add me please @haroune_foo